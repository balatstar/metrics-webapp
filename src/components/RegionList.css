.region-title {
  height: 175px;
  display: flex;
  background-color: #fb5192;
}

.region-title > * {
  width: 50%;
}

.region-map {
  background: url(images/united-kingdom.svg) center center no-repeat;
  background-size: contain;
}

.region-title .low {
  background: url(images/icon-low.svg) center center/100px no-repeat;
}

.region-title .high {
  background: url(images/icon-high.svg) center center/100px no-repeat;
}

.region-title .moderate {
  background: url(images/icon-moderate.svg) center center/100px no-repeat;
}

.region-title .very {
  background-size: 100px;
  background-repeat: repeat-x;
  background-position: left;
}

.region-title h2 {
  padding-left: 20px;
  padding-top: 60px;
  text-transform: uppercase;
  font-size: 16px;
}

.region-head {
  background-color: #df4782;
}

.region-head h3 {
  margin: 0;
  padding: 5px;
  font-size: 14px;
  font-weight: 500;
  text-transform: uppercase;
}

.region-content {
  list-style: none;
  margin: 0;
  padding: 0;
}

.region-content-links {
  display: flex;
  justify-content: space-between;
  padding: 20px 10px;
}

.region-content-links:nth-child(odd) {
  background-color: #e94987;
}

.region-content-links:nth-child(even) {
  background-color: #cb4175;
}

.region-content-links > * {
  width: 47%;
}

.region-next {
  width: 4%;
}

.region-name {
  font-weight: 500;
  font-size: 18px;
}

.region-name a,
.region-details a {
  color: #fff;
  text-decoration: none;
}

.region-details {
  padding-right: 5px;
  text-transform: capitalize;
  text-align: right;
}
