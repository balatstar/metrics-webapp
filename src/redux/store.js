import { configureStore, combineReducers } from '@reduxjs/toolkit';
import regionReducer from '../redux/region/regionSlice';

const rootReducer = combineReducers({
  regions: regionReducer,
});

const initialState = {
  regions: {
    regions: [],
    loading: 'idle',
    error: null,
    uniqueIntensityIndexValues: [],
    lowestForecastValues: {},
  },
};

const store = configureStore({
  reducer: rootReducer,
  preloadedState: initialState,
});

export default store;
