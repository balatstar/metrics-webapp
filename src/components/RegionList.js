import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRegions } from '../redux/region/regionSlice';
import { Link, useParams } from 'react-router-dom';
import './RegionList.css';
import arrow from './images/right-arrow.png';

const RegionList = () => {
  const { intensityIndex } = useParams();
  const dispatch = useDispatch();
  const regions = useSelector((state) => state.regions.regions);
  const loading = useSelector((state) => state.regions.loading);
  const error = useSelector((state) => state.regions.error);

  useEffect(() => {
    dispatch(fetchRegions());
  }, [dispatch]);

  if (loading === 'loading') {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error}</div>;
  }

  const filteredRegions = intensityIndex
    ? regions.filter((region) => region.intensity.index === intensityIndex)
    : regions;

  return (
    <div>
      <div className="region-title">
        <div className={`region-map ${intensityIndex}`}></div>
        {intensityIndex ? (
          <h2>{intensityIndex}</h2>) : (<h2>All Regions</h2>
        )}
      </div>
      <div className="region-head"><h3>Region Summary</h3></div>
      <ul className="region-content">
        {filteredRegions.map((region) => (
          <li className="region-content-links" key={region.regionid}>
            <div className='region-name'>
              <Link to={`/region/${region.regionid}`}>{region.shortname}</Link>
            </div>
            <div className='region-details'>
              <Link to={`/region/${region.regionid}`}>
                Intensity Forecast: {region.intensity.forecast}<br />
                {!intensityIndex && (
                  <span>{region.intensity.index}</span>
                )}
              </Link>
            </div>
            <div className="region-next"><Link to={`/region/${region.regionid}`}><img src={arrow} /></Link></div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default RegionList;
