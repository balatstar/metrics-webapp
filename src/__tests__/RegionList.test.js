import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import RegionList from '../components/RegionList';

describe('RegionList', () => {
  it('renders RegionList component', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <RegionList />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
