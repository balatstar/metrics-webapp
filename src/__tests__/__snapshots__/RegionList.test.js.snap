// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionList renders RegionList component 1`] = `
<div>
  <div>
    Loading...
  </div>
</div>
`;
